### PR TITLE
Channel and PayloadChannel: Optimize message format and JSON generation

### PR DESCRIPTION
**Pull request by ibc**
- Replaces PR #870.

### TODO

- Rust layer.

### CODE REVIEW

**Comment by ibc** [comment]
Everything seems to work. I'm gonna merge this.

**Comment by ibc** [comment]
However, changes are needed in Rust and definitely I don't know how to apply the required changes in `channel.rs`, `payload_channel.rs`, `data_producer.rs` and `data_consumer.rs`.

@nazar-pc could you please do them in a separate PR?

**Comment by nazar-pc** [comment]
Doesn't seem like I have a choice. This might have to wait until Saturday on my end, but doesn't look complicated at all and I might get it done sooner.

**Comment by ibc** [comment]
Unfortunately Jose and me need changes in this PR before Friday, that's why I couldn't wait for Rudy changes 😞

**Review by jmillan — COMMENTED** [review]
:+1:

File: worker/src/Channel/ChannelSocket.cpp [hunk 1]
```diff
@@ -171,42 +189,6 @@ namespace Channel
 		auto free = this->channelReadFn(
 		  &message, &messageLen, &messageCtx, this->uvReadHandle, this->channelReadCtx);
 
-		if (free)
```
Comment (nazar-pc) [inline worker/src/Channel/ChannelSocket.cpp:174]: Why did you delete this? This was the way Rust communicated with C++ worker.

File: worker/src/PayloadChannel/PayloadChannelSocket.cpp [hunk 2]
```diff
@@ -196,94 +224,6 @@ namespace PayloadChannel
 		  this->uvReadHandle,
 		  this->payloadChannelReadCtx);
 
-		if (free)
```
Comment (nazar-pc) [inline worker/src/PayloadChannel/PayloadChannelSocket.cpp:199]: And the same here, you just dropped Rust support, while it had no effect on TypeScript

**Comment by nazar-pc** [comment]
I see, it doesn't compile with those. But it would have been nice to fix compilation instead of deleting the code altogether :disappointed:

**Comment by ibc** [comment]
I think Jose commented that code in his ongoing PR, I didn't know what it was for before and removed it. Sorry.

**Comment by nazar-pc** [comment]
I will restore it in upcoming PR in a separate commit.
